### PR TITLE
fix(sqllab): per-tab hide left bar

### PR DESCRIPTION
**Pull request by betodealmeida**
### SUMMARY

Currently the state of whether we should show the left bar in SQL Lab is **global**, not per tab. Since this is configured separately in each tab menu, it makes sense to track the state individually.

This PR moves the state of the left bar to the query editor, using the redux store to toggle its display. If the `SQLLAB_BACKEND_PERSISTENCE` is enabled the visibility is stored on the backend as well.

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF

N/A

### TEST PLAN

0. `SQLLAB_BACKEND_PERSISTENCE` set to off.
1. Created a few tabs, changed their visibility, verified that visibility is defined per tabs and persists on page refresh.
2. Turn `SQLLAB_BACKEND_PERSISTENCE` on, reload SQL Lab, verified that tabs' state was migrated to the backend.
3. Toggle visibility, verified state is persisted in the backend and loaded correctly on page refresh and tab switch.

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Changes UI
- [ ] Requires DB Migration.
- [ ] Confirm DB Migration upgrade and downgrade tested.
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: superset-frontend/src/SqlLab/components/SqlEditor.jsx [hunk 1]
```diff
@@ -710,14 +710,17 @@ class SqlEditor extends React.PureComponent {
         ? 'Specify name to CREATE VIEW AS schema in: public'
         : 'Specify name to CREATE TABLE AS schema in: public';
 
+    const leftBarStateClass = this.props.hideLeftBar
+      ? 'schemaPane-exit-done'
+      : 'schemaPane-enter-done';
```
Comment (betodealmeida) [inline superset-frontend/src/SqlLab/components/SqlEditor.jsx:715]: Looks like there's a bug in the library: https://github.com/reactjs/react-transition-group/issues/643

Without this, if `this.props.hideLeftBar` was initially set to true the bar wouldn't hide on page load.

File: superset-frontend/src/SqlLab/actions/sqlLab.js [hunk 2]
```diff
@@ -663,6 +664,36 @@ export function setActiveSouthPaneTab(tabId) {
   return { type: SET_ACTIVE_SOUTHPANE_TAB, tabId };
 }
 
+export function toggleLeftBar(queryEditor) {
+  const hideLeftBar = !queryEditor.hideLeftBar;
+  return function (dispatch) {
+    const sync = isFeatureEnabled(FeatureFlag.SQLLAB_BACKEND_PERSISTENCE)
+      ? SupersetClient.put({
```
Comment (hughhhh) [inline superset-frontend/src/SqlLab/actions/sqlLab.js:671]: can we use `makeApi` here instead of `SupersetClient` directly

Comment (betodealmeida) [reply superset-frontend/src/SqlLab/actions/sqlLab.js:671]: @hughhhh I'd rather leave this consistent with the rest of the code that manages `SQLLAB_BACKEND_PERSISTENCE`, if that's ok, until we have time to refactor the whole thing. Then we could maybe create a custom hook to manage the whole state.

**Review by hughhhh — APPROVED** [review]

**Comment by ktmud** [comment]
@betodealmeida @hughhhh There seems to be a merge conflict in db-migration: https://github.com/apache/superset/runs/2042369464

**Pull request merged**